### PR TITLE
Device_doctor: add healthcheck to android device

### PR DESCRIPTION
**Pull request by keyonghan**
`healthcheck` was conducted but it failed to analyze results and communicate with console for android device. This PR adds it.

This PR also removes unnecessary `throw`.

### CODE REVIEW

File: device_doctor/lib/src/health.dart [hunk 1]
```diff
@@ -88,14 +88,12 @@ class HealthCheckResult {
...
   if (deviceChecks.isEmpty) {
     stderr.writeln('No healthy device is available');
-    throw StateError('No healthy device is available');
   }
   for (String deviceID in deviceChecks.keys) {
     List<HealthCheckResult> checks = deviceChecks[deviceID];
     for (HealthCheckResult healthCheckResult in checks) {
       if (!healthCheckResult.succeeded) {
         stderr.writeln('${healthCheckResult.name} check failed with: ${healthCheckResult.details}');
-        throw StateError('$deviceID: ${healthCheckResult.name} check failed with: ${healthCheckResult.details}');
```
Comment (CaseyHillers) [inline device_doctor/lib/src/health.dart:98]: Why is this removed? How else does device_doctor indicate to the bot config the health checks failed?

Comment (keyonghan) [reply device_doctor/lib/src/health.dart:98]: It is based on process.communicate, which uses stdout and stderr message.

**Review by CaseyHillers — APPROVED** [review]
